doc/src/examples/dot_product

```diff
@@ -0,0 +1,66 @@
+// Boost.Geometry
```

Comment:
Please, remove this extension-less file `doc/src/examples/dot_product` which seems to be copy of the correct one `doc/src/examples/arithmetic/dot_product.cpp`.